src/io/flutter/utils/IconPreviewGenerator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.utils;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.util.TripleFunction;
+import io.flutter.FlutterUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.font.FontRenderContext;
+import java.awt.font.LineMetrics;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Rectangle2D;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.util.Properties;
+
+@SuppressWarnings("UseJBColor")
+public class IconPreviewGenerator {
+  private static final Logger LOG = Logger.getInstance(IconPreviewGenerator.class);
+
+  @NotNull final String fontFilePath;
+  int iconSize = 16;
+  int fontSize = 16;
+  @NotNull Color fontColor = Color.gray;
+
+  public IconPreviewGenerator(@NotNull String fontFilePath) {
+    this.fontFilePath = fontFilePath;
+  }
+
+  public IconPreviewGenerator(@NotNull String fontFilePath, int iconSize, int fontSize, @Nullable Color fontColor) {
+    this(fontFilePath);
+    this.iconSize = iconSize;
+    this.fontSize = fontSize;
+    if (fontColor != null) {
+      this.fontColor = fontColor;
+    }
+  }
+
+  public Icon convert(String hexString) {
+    if (hexString.startsWith("0x") || hexString.startsWith("0X")) {
+      hexString = hexString.substring(2);
+    }
+    return convert(Integer.parseInt(hexString, 16));
+  }
+
+  public Icon convert(int code) {
```

Comment:
When is this single icon convert meant to be used? If the end user has a color hex in their code? If possible it would be nice to have a test for this method as well.

Reply:
Yes, this is the entry-point for the plugin to use while IntelliJ is walking the Dart AST. We'd look up the name if an icon was referenced by name, to get the hex number, or get the number from the AST if it was referenced by codepoint.